internal/exec/stages/disks/partitions.go

```diff
@@ -353,7 +354,14 @@ func (s stage) partitionDisk(dev types.Disk, devAlias string) error {
 		case exists && shouldExist && matches:
 			s.Logger.Info("partition %d found with correct specifications", part.Number)
 		case exists && shouldExist && !wipeEntry && !matches:
-			return fmt.Errorf("Partition %d didn't match: %v", part.Number, matchErr)
+			if resize {
+				s.Logger.Info("partition %d found with resizePartition option, wiping partition entry and recreating", part.Number)
```

Comment:
`wiping partition entry and recreating` may be technically true but I don't think it conveys the correct impression to the user.  From their perspective it's a resize.